parser/src/test/java/com/linkedin/avroutil1/util/ConfigurableAvroSchemaComparatorTest.java

```diff
@@ -22,23 +27,89 @@ public class ConfigurableAvroSchemaComparatorTest {
   @DataProvider
   private Object[][] testEqualsProvider() {
     return new Object[][] {
-        {"schemas/TestRecord.avsc", "schemas/TestRecord.avsc", true}
+        {"schemas/TestRecord.avsc", "schemas/TestRecord.avsc"}
     };
   }
 
-  @Test(dataProvider = "testEqualsProvider")
-  public void testEquals(String path1, String path2, boolean isEqual) throws IOException {
+  private static AvroSchema validateAndGetAvroRecordSchema(String path) throws IOException {
     AvscParser parser = new AvscParser();
-    String avsc1 = TestUtil.load(path1);
-    String avsc2 = TestUtil.load(path2);
-    AvscParseResult result1 = parser.parse(avsc1);
-    AvscParseResult result2 = parser.parse(avsc2);
-    Assert.assertNull(result1.getParseError());
-    Assert.assertNull(result2.getParseError());
-    AvroRecordSchema recordSchema1 = (AvroRecordSchema) result1.getTopLevelSchema();
-    AvroRecordSchema recordSchema2 = (AvroRecordSchema) result2.getTopLevelSchema();
-    Assert.assertTrue(
-        ConfigurableAvroSchemaComparator.equals(recordSchema1, recordSchema2, SchemaComparisonConfiguration.PRE_1_7_3));
+    String avsc = TestUtil.load(path);
+    AvscParseResult result = parser.parse(avsc);
+    Assert.assertNull(result.getParseError());
+    return result.getTopLevelSchema();
+  }
+
+  @Test(dataProvider = "testEqualsProvider")
+  public void testEquals(String path1, String path2) throws IOException {
+    AvroRecordSchema recordSchema1 = (AvroRecordSchema) validateAndGetAvroRecordSchema(path1);
+    AvroRecordSchema recordSchema2 = (AvroRecordSchema) validateAndGetAvroRecordSchema(path2);
+    Assert.assertTrue(ConfigurableAvroSchemaComparator.equals(recordSchema1, recordSchema2, SchemaComparisonConfiguration.PRE_1_7_3));
+  }
+
+  @Test
+  public void testNotEquals() throws IOException {
+    AvroRecordSchema schema1 = (AvroRecordSchema) validateAndGetAvroRecordSchema("schemas/UtilTester1.avsc");
+    AvroRecordSchema schema2 = (AvroRecordSchema) validateAndGetAvroRecordSchema("schemas/UtilTester2.avsc");
+    Assert.assertFalse(ConfigurableAvroSchemaComparator.equals(schema1, schema2, SchemaComparisonConfiguration.PRE_1_7_3));
+  }
```

Comment:
why not use dataProvider on all of these test methods? these methods are very similar, so you can just have one method, but use dataProvider to provide different inputs to the test method.

Reply:
Thanks for the callout, added the change in the most recent commit